Jenkinsfile

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env groovy
+
+def mainBranches() {
+    return BRANCH_NAME == "develop" || BRANCH_NAME.startsWith("release/");
+}
+
+// TODO: Use multiple choices
+run_linter = true
+rust_test = true
+helm_test = true
+run_tests = params.run_tests
+
+// Will skip steps for cases when we don't want to build
+if (currentBuild.getBuildCauses('jenkins.branch.BranchIndexingCause') && mainBranches()) {
+    print "INFO: Branch Indexing, skip tests and push the new images."
+    run_tests = false
+}
+
+// Only schedule regular builds on main branches, so we don't need to guard against it
+String cron_schedule = mainBranches() ? "0 2 * * *" : ""
+
+pipeline {
+  agent none
+  options {
+    timeout(time: 1, unit: 'HOURS')
```

Comment:
Maybe change to 2 hours.
I've already made this change in control-plane as now that we have loads of nightly runs the agents get busy and it takes longer to get a free agent to run on.
We should probably consider adding more agents to Jenkins.

Reply:
Done.